newIDE/app/src/ResourcesList/LocalYarnBridge.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+import optionalRequire from '../Utils/OptionalRequire.js';
+import { type ExternalEditorOpenOptions } from './ResourceExternalEditor.flow';
+import {
+  createOrUpdateResource,
+  getLocalResourceFullPath,
+} from './ResourceUtils.js';
+
+const electron = optionalRequire('electron');
+const path = optionalRequire('path');
+const ipcRenderer = electron ? electron.ipcRenderer : null;
+const gd = global.gd;
+
+/**
+ * Open YARN to Create/Edit Json Dialogue Tree resources.
+ */
+export const openYarn = ({
+  project,
+  resourcesLoader,
+  resourceNames,
+  onChangesSaved,
+  resourcePath,
+  extraOptions,
+}: ExternalEditorOpenOptions) => {
+  if (!electron || !ipcRenderer) return;
+  const projectPath = path.dirname(project.getProjectFile());
+  const initialResourcePath = getLocalResourceFullPath(
+    project,
+    resourceNames[0]
+  );
+
+  const externalEditorData = {
```

Comment:
Let's not change it now, but I'm not a fan of calling an object externalEditorData and have inside something called... externalEditorData. Definitely a code smell.
Let's not change it for now but could have been named differently, like "yarnArguments" or something like this.

Reply:
ok, will leave it for another pr :)